tests/unit/src/Internal/Payment/State/ProcessedStateTest.php

```diff
@@ -93,8 +101,24 @@ public function test_complete_processing_will_transition_to_completed_state() {
 			->with( CompletedState::class, $this->mock_context )
 			->willReturn( $mock_completed_state );
 
+		$this->mock_legacy_proxy
+			->expects( $this->once() )
+			->method( 'call_function' )
+			->with( 'wc' )
+			->willReturnCallback(
+				function () {
+					$mock_cart = $this->getMockBuilder( \stdClass::class )
+						->setMethods( [ 'empty_cart' ] )
+						->getMock();
+					return (object) [
+						'cart' => $mock_cart,
+					];
+				}
+			);
+
 		$result = $this->sut->complete_processing();
 
 		$this->assertSame( $mock_completed_state, $result );
 	}
+
```

Comment:
Please also add tests for `clear_cart` and `clear_upe_payment_intent_from_session`. They are marked as covered, because they get called, but we don't actually tests whether they do what they do.

Reply:
Should we add a check for whether `clear_upe_payment_intent_from_session` is called, or add test whether the session has been cleared. Latter is less of unit test, isn't it?

Reply:
The former: We just need to make sure it is called. Currently, if we remove the call to the method, the test will not fail.